### PR TITLE
[WIP] chore: update module federation config

### PR DESCRIPTION
**Pull request by vkbansal**
##### Screenshots:

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

File: configs/modulefederation.config.js [hunk 1]
```diff
@@ -7,20 +7,19 @@
 
 const packageJSON = require('../package.json')
 const { pick, omit, mapValues } = require('lodash')
+const { version } = require('fork-ts-checker-webpack-plugin')
```
Comment (knagurski) [inline configs/modulefederation.config.js:10]: You're not actually using this

File: configs/modulefederation.config.js [hunk 2]
```diff
@@ -52,7 +51,8 @@ module.exports = ({ enableGitOpsUI, enableSTO }) => {
       mapValues(pick(packageJSON.dependencies, ExactSharedPackages), version => ({
         singleton: true,
         requiredVersion: version
-      }))
+      })),
+      pick(packageJSON.dependencies, SharedPackages)
```
Comment (knagurski) [inline configs/modulefederation.config.js:55]: I was thinking something more like this...
```typescript
      shared: {
        ...packageJSON.dependencies,
        ...ExactSharedPackages.reduce((deps, exactDepName) => ({
          ...deps,
          [exactDepName]: {
            singleton: true,
            requiredVersion: packageJSON.dependencies[exactDepName]
          }
        }))
      }
```

File: configs/modulefederation.config.js [hunk 3]
```diff
@@ -7,20 +7,19 @@
...
 const { pick, omit, mapValues } = require('lodash')
+const { version } = require('fork-ts-checker-webpack-plugin')
 
 /**
  * These packages must be stricly shared with exact versions
  */
-const ExactSharedPackages = [
+const ExactSharedPackages = ['react-dom', 'react', 'react-router-dom', 'restful-react', '@harness/use-modal']
+
+const SharedPackages = [
```
[the remark this answers is not in the document]
Comment (scottyw) [reply configs/modulefederation.config.js:17]: I could be missing your point here, but do you mean it doesn't seem valuable to have 2 lists or that it doesn't seem valuable to share thing explicitly vs sharing everything by default?

If the former, I think the goal is to separate what _must_ be shared (singletons) from what is shared for optimization reasons. The required shared stuff should be identical for all MFEs and unambiguously must be shared, so is correspondingly difficult to upgrade in a breaking way. The optionally shared things can be unshared for upgrade purposes i.e. we can unshare and upgrade independently of platform. Then we would reshare once our MFE and platform are both upgraded. That fixes the deployment problem and helps us understand what we can upgrade safely and what we can't.

If the latter, I think we need to be conscious in the MFE about our choice to share the platform version of a dep. We want to do that as much as possible but each dependency we inherit is a potential source of breakage that we need to manage. Explicit sharing lets us make that decision and take on that cost in a conscious way, and prevents new dependencies catching us out.

[the remark this answers is not in the document]
Comment (vkbansal) [reply configs/modulefederation.config.js:17]: We started out with sharing all the dependencies, but it was removed as it causing some issues in OPA https://github.com/harness/harness-core-ui/commit/8f5d2258bbde0fbd2a3f79c1fa586312edec4d3f

File: configs/modulefederation.config.js (hunk 2 above)
Comment (vkbansal) [reply configs/modulefederation.config.js:55]: We started out with sharing all the dependencies, but it was removed as it causing some issues in OPA https://github.com/harness/harness-core-ui/commit/8f5d2258bbde0fbd2a3f79c1fa586312edec4d3f

**Comment by scottyw** [comment]
Will these changes break existing MFE code? I'm thinking about deployment.

**Pull request closed without merging**